File: CabinApi/DTOs/RegisterDTO.cs
namespace CabinApi.DTOs
{
    public class RegisterDTO
    {
        public string FullName { get; set; } = string.Empty;
        public string Username { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public string PostalCode { get; set; } = string.Empty;
        public int Role { get; set; } = 0;
        public DateTime CreatedDate { get; set; } = DateTime.Now;
        public DateTime ModifiedDate { get; set; } = DateTime.Now;
        public byte IsActive { get; set; } = 0;
    }
}

File: cabin-app/src/DTO/RegisterDTO.js
export const registerDO = {
  fullName: {
    type: "text",
    value: "",
    errors: "",
  },
  username: {
    type: "text",
    value: "",
    errors: "",
  },
  email: {
    type: "email",
    value: "",
    errors: "",
  },
  emailConfirm: {
    type: "email",
    value: "",
    errors: "",
  },
  password: {
    type: "password",
    value: "",
    errors: "",
  },
  passwordConfirm: {
    type: "password",
    value: "",
    errors: "",
  },
  countryCode: {
    type: "tel",
    value: "+358",
    errors: "",
  },
  phone: {
    type: "tel",
    value: "",
    errors: "",
  },
  address: {
    type: "text",
    value: "",
    errors: "",
  },
  postalCode: {
    type: "text",
    value: "",
    errors: "",
  },
  role: {
    type: "option",
    value: "0",
    errors: "",
    0: {
      name: "Role",
    },
    1: {
      name: "Admin",
    },
    2: {
      name: "Manager",
    },
    3: {
      name: "Supervisor",
    },
    4: {
      name: "Worker",
    },
  },
  termsOfService: {
    type: "checkbox",
    isActive: false,
    value: false,
    0: {
      name: "I agree to the terms and conditions of the Service",
    },
  },
};


export const registerDTO = {
  fullName: "",
  username: "",
  email: "",
  emailConfirm: "",
  password: "",
  passwordConfirm: "",
  phone: "",
  address: "",
  postalCode: "",
  role: "",
  createdDate: "",
  modifiedDate: "",
  isActive: 0,
};